test/upgrade/postupgrade.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"os"
+	"testing"
+
+	mf "github.com/manifestival/manifestival"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
+	"knative.dev/operator/pkg/reconciler/common"
+	util "knative.dev/operator/pkg/reconciler/common/testing"
+	"knative.dev/operator/test"
+	"knative.dev/operator/test/client"
+	"knative.dev/operator/test/resources"
+	pkgupgrade "knative.dev/pkg/test/upgrade"
+)
+
+// OperatorPostUpgradeTests verifies the KnativeServing and KnativeEventing creation, deployment recreation, and
+// the deletion after the operator upgrades to the latest release.
+func OperatorPostUpgradeTests() []pkgupgrade.Operation {
+	return []pkgupgrade.Operation{
+		ServingPostUpgradeTests(),
+		EventingPostUpgradeTests(),
```

Comment:
Why this one doesn't reuse upstream operations as well, or solely?

Reply:
Besides the existing serving and eventing upgrade and downgrade tests, Operator itself has the upgrade and downgrade tests, including checking the status of the CRs and deployments.
So ServingPostUpgradeTests and EventingPostUpgradeTests are calling tests in the operator repo.
There are three upgrade prows in the operator.
`upgrade-tests`: calling upgrade and downgrade tests in the operator.
`serving-upgrade-tests`: calling serving tests in serving repo and the serving tests in the operator.
`eventing-upgrade-tests`: eventing only, calling eventing tests in eventing repo and the eventing tests in the operator.

Reply:
Okay. I understand that. That should be documented in some README file - please add one in `test/upgrade/README.md`.

Also, I would like to see a run that contains both serving & eventing. Eventing upgrade tests contains an optional integration with serving that I long wanted to enable here.

Reply:
@cardil We used to run them together, enabling E2E_UPGRADE_TESTS_SERVING_USE, but it always fails, so I separate them. I have no clue why it failed, do not want to block the dev with failing prow.

Reply:
Right. There is issue that tracks that: knative/operator#297

After we merge this PR, I think I'll take a look at it, to debug what is happening and add this interop tests in one of current prow jobs of maybe in new one.

So, lets move the interop (E2E_UPGRADE_TESTS_SERVING_USE flag) to knative/operator#297.

I was also asking, why not to add upstream operations here as well. They should interfere with each other. We can also tacle that in knative/operator#297, if you like.

Reply:
Three prows are now working like this:
upgrade-tests: installs serving and eventing CRs, and run the operator CR tests, serving CR and eventing CR.
serving-upgrade-tests: installs serving and eventing CRs, and run the tests of serving CR and serving.
eventing-upgrade-tests: installs serving and eventing CRs, and run the tests of eventing CR and eventing.

This is not what you expected?

Reply:
Let's go with that setup for now.

When I get to fixing #297 I'll try to make it as follows:

upgrade-tests: installs serving and eventing CRs, and run the operator CR tests, serving CR, upstream serving and eventing CR and upstream eventing with interop flags enabled.
serving-upgrade-tests: installs serving and eventing CRs, and run the tests of serving CR and serving.
eventing-upgrade-tests: installs serving and eventing CRs, and run the tests of eventing CR and eventing.